### ISSUE TITLE
Test failure: AppCanHandleClientAbortingConnectionMidRequest

### ISSUE THREAD

**Issue by aspnet-hello** [aspnet/KestrelHttpServer#2980]
This issue was made automatically. If there is a problem contact ryanbrandenburg.

This test [failed](http://aspnetci/viewLog.html?buildId=567511&buildTypeId=Releases_22xPublic_UbuntuUniverse) with the following error:

```
Moq.MockException : 
Expected invocation on the mock once, but was 0 times: t => t.ConnectionStop(It.IsAny<String>())
No setups configured.
Performed invocations:
ILogger.IsEnabled(Information)
ILogger.Log<Object>(Debug, 0, [[{OriginalFormat}, TestServer is listening on port 35410]], null, System.Func`3[System.Object,System.Exception,System.String])
IKestrelTrace.ConnectionStart("0HLH9CKIPJ3CI")
ILogger.IsEnabled(Critical)
ILogger.BeginScope<ConnectionLogScope>([[ConnectionId, 0HLH9CKIPJ3CI]])
IKestrelTrace.RequestBodyStart("0HLH9CKIPJ3CI", "0HLH9CKIPJ3CI:00000001")
IKestrelTrace.ConnectionDisconnect("0HLH9CKIPJ3CI")
IKestrelTrace.RequestBodyDone("0HLH9CKIPJ3CI", "0HLH9CKIPJ3CI:00000001")
IKestrelTrace.ApplicationError("0HLH9CKIPJ3CI", "0HLH9CKIPJ3CI:00000001", Microsoft.AspNetCore.Connections.ConnectionResetException: Error -104 ECONNRESET connection reset by peer ---> Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.Internal.Networking.UvException: Error -104 ECONNRESET connection reset by peer
   --- End of inner exception stack trace ---
   at System.IO.Pipelines.PipeCompletion.ThrowLatchedException()
   at System.IO.Pipelines.PipeCompletion.IsCompletedOrThrow()
   at System.IO.Pipelines.Pipe.GetReadResult(ReadResult& result)
   at System.IO.Pipelines.Pipe.GetReadAsyncResult()
   at System.IO.Pipelines.Pipe.DefaultPipeReader.GetResult(Int16 token)
   at System.Threading.Tasks.ValueTask`1.get_Result()
   at Microsoft.AspNetCore.Server.Kestrel.Core.Adapter.Internal.RawStream.ReadAsyncInternal(Memory`1 destination) in /_/src/Kestrel.Core/Adapter/Internal/RawStream.cs:line 123
   at Microsoft.AspNetCore.Server.Kestrel.Core.Adapter.Internal.AdaptedPipeline.ReadInputAsync(Stream stream) in /_/src/Kestrel.Core/Adapter/Internal/AdaptedPipeline.cs:line 135
   at System.IO.Pipelines.PipeCompletion.ThrowLatchedException()
   at System.IO.Pipelines.PipeCompletion.IsCompletedOrThrow()
   at System.IO.Pipelines.Pipe.GetReadResult(ReadResult& result)
   at System.IO.Pipelines.Pipe.ReadAsync(CancellationToken token)
   at System.IO.Pipelines.Pipe.DefaultPipeReader.ReadAsync(CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.Http1MessageBody.PumpAsync() in /_/src/Kestrel.Core/Internal/Http/Http1MessageBody.cs:line 120
   at System.IO.Pipelines.PipeCompletion.ThrowLatchedException()
   at System.IO.Pipelines.PipeCompletion.IsCompletedOrThrow()
   at System.IO.Pipelines.Pipe.GetReadResult(ReadResult& result)
   at System.IO.Pipelines.Pipe.ReadAsync(CancellationToken token)
   at System.IO.Pipelines.Pipe.DefaultPipeReader.ReadAsync(CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.MessageBody.CopyToAsync(Stream destination, CancellationToken cancellationToken) in /_/src/Kestrel.Core/Internal/Http/MessageBody.cs:line 86
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpRequestStream.CopyToAsyncInternal(Stream destination, CancellationToken cancellationToken) in /_/src/Kestrel.Core/Internal/Http/HttpRequestStream.cs:line 157
   at Microsoft.AspNetCore.Server.Kestrel.FunctionalTests.RequestTests.<>c__DisplayClass21_0.<<AppCanHandleClientAbortingConnectionMidRequest>b__1>d.MoveNext() in /_/test/Kestrel.Transport.FunctionalTests/RequestTests.cs:line 834
--- End of stack trace from previous location where exception was thrown ---
   at Xunit.Assert.RecordExceptionAsync(Func`1 testCode) in C:\Dev\xunit\xunit\src\xunit.assert\Asserts\Record.cs:line 82
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequests[TContext](IHttpApplication`1 application) in /_/src/Kestrel.Core/Internal/Http/HttpProtocol.cs:line 532)
   at Moq.Mock.ThrowVerifyException(MethodCall expected, IEnumerable`1 setups, IEnumerable`1 actualCalls, Expression expression, Times times, Int32 callCount)
   at Moq.Mock.VerifyCalls(Interceptor targetInterceptor, MethodCall expected, Expression expression, Times times)
   at Moq.Mock.Verify[T](Mock`1 mock, Expression`1 expression, Times times, String failMessage)
   at Moq.Mock`1.Verify(Expression`1 expression, Times times)
   at Microsoft.AspNetCore.Server.Kestrel.FunctionalTests.RequestTests.AppCanHandleClientAbortingConnectionMidRequest(ListenOptions listenOptions) in /_/test/Kestrel.Transport.FunctionalTests/RequestTests.cs:line 865
--- End of stack trace from previous location where exception was thrown ---
------- Stdout: -------
| [0.002s] TestLifetime Information: Starting test AppCanHandleClientAbortingConnectionMidRequest-http://127.0.0.1:0 at 2018-10-03T19:19:32
| [0.003s] Microsoft.AspNetCore.Hosting.Internal.WebHost Debug: Hosting starting
| [0.015s] Microsoft.AspNetCore.Hosting.Internal.WebHost Debug: Hosting started
| [0.016s] Microsoft.AspNetCore.Hosting.Internal.WebHost Debug: Loaded hosting startup assembly Libuv.FunctionalTests, Version=2.2.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60
| [0.016s] Microsoft.AspNetCore.Server.Kestrel Debug: TestServer is listening on port 35410
| [0.016s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "0HLH9CKIPJ3CI" started.
| [0.018s] Microsoft.AspNetCore.Hosting.Internal.WebHost Information: Request starting HTTP/1.1 POST http:///  8192
| [0.019s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "0HLH9CKIPJ3CI", Request id "0HLH9CKIPJ3CI:00000001": started reading request body.
| [0.023s] Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv Debug: Connection id "0HLH9CKIPJ3CI" reset.
| [0.024s] Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv Debug: Connection id "0HLH9CKIPJ3CI" sending FIN.
| [0.024s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "0HLH9CKIPJ3CI" disconnecting.
| [0.031s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "0HLH9CKIPJ3CI", Request id "0HLH9CKIPJ3CI:00000001": done reading request body.
| [0.038s] Microsoft.AspNetCore.Hosting.Internal.WebHost Debug: Hosting shutdown
| [0.
...
```

Other tests within that build may have failed with a similar message, but they are not listed here. Check the link above for more info.

This test failed on 2.2.

CC @muratg

**Comment by aspnet-hello** [comment]
This comment was made automatically. If there is a problem contact ryanbrandenburg.

Please use this workflow to address this flaky test issue, including checking applicable checkboxes and filling in the applicable "TODO" entries:

* Is this actually a flaky test?
  * No, this is a regular test failure, fix the test/product (TODO: Link to commit/PR)
  * Yes, proceed below...

* Is this test failure caused by product code flakiness? (Either this product, or another product this test depends on.)
  * [ ] File a bug against the product (TODO: Link to other bug)
  * Is it possible to change the test to avoid the flakiness?
    * Yes? Go to the "Change the test!" section.
    * No?
      * [ ] Disable the test (TODO: Link to PR/commit)
      * [ ] Wait for other bug to be resolved
      * [ ] Wait for us to get build that has the fix
      * [ ] Re-enable our test (TODO: Link to PR/commit)
      * [ ] Close this bug

* Is it that the test itself is flaky? This includes external transient problems (e.g. remote server problems, file system race condition, etc.)
  * Is there is a way to change our test to avoid this flakiness?
    * Yes? Change the test!
      * [ ] Change the test to avoid the flakiness, for example by using a different test strategy, or by adding retries w/ timeouts (TODO: Link to PR/commit)
      * [ ] Run the test 100 times locally as a sanity check.
      * [ ] Close this bug
    * No?
      * Is there any logging or extra information that we could add to make this more diagnosable when it happens again?
        * Yes?
            * [ ] Add the logging (TODO: Link to PR/commit)
        * No?
            * [ ] Delete the test because flaky tests are not useful (TODO: Link to PR/commit)

**Comment by muratg** [comment]
cc @halter73 could you take a look?

**Comment by halter73** [comment]
This should be fixed with #2977
